pages/members.js

```diff
@@ -0,0 +1,221 @@
+import React from 'react';
+import styled from 'styled-components';
+import { gql } from 'apollo-boost';
+import { useQuery } from '@apollo/react-hooks';
+import { Grid, Cell } from 'styled-css-grid';
+import { NextSeo } from 'next-seo';
+import _ from 'lodash';
+import ContentSection from '../components/shared/ContentSection';
+import LinkButton from '../components/shared/LinkButton/LinkButton';
+import SquareButton from '../components/shared/SquareButton';
+import SkeletonLoader from '../components/shared/SkeletonLoader';
+import LoadingIndicator from '../components/shared/LoadingIndicator';
+import ProfileItem from '../components/shared/ProfileItem';
+import { HeroGraphicDiv } from '../components/shared/StandardStyles';
+import { below, gridRepeat } from '../utilities';
+
+const GET_MEMBERS = gql`
+  query getMembers($after: String) {
+    members {
+      members(after: $after, orderBy: FIRSTNAME, pageSize: 20) {
+        cursor
+        members {
+          id
+          firstName
+          lastName
+          profileSlug
+          jobTitle
+          company
+          profileImage
+        }
+      }
+    }
+  }
+`;
+
+const StyledHeroGraphicDiv = styled(HeroGraphicDiv)`
+  text-align: right;
+`;
+
+const HighlightImage = styled.img`
+  max-height: 35rem;
+  object-fit: contain;
+`;
+
+const MembersContentSection = styled(ContentSection)`
+  padding-top: 0;
+`;
+
+const ProfileSection = styled.div`
+  display: flex;
+  margin: auto;
+  justify-content: center;
+  flex-wrap: wrap;
+  max-width: 120rem;
+
+  ${below.small`
+    margin-top: 2rem;
+  `};
+`;
+
+const StyledProfileItem = styled(ProfileItem)`
+  margin: 3rem 1rem 0;
+  width: 22rem;
+
+  ${below.small`
+    width: 15rem;
+  `};
+`;
+
+const LoadMoreButton = styled(SquareButton)`
+  min-width: 20rem;
+  height: 6rem;
+  display: block;
+  margin-top: 4rem;
+  margin-left: auto;
+  margin-right: auto;
+
+  &:focus {
+    outline: unset;
+  }
+
+  ${below.small`
+    width: 100%;
+  `};
+`;
+
+let fetchingMore = false;
+
+const memberListing = () => {
+  // networkStatus and notifyOnNetworkStatusChange are needed to know when fetching additional members.
+  const { loading, error, data, fetchMore, networkStatus } = useQuery(
+    GET_MEMBERS,
+    {
+      notifyOnNetworkStatusChange: true,
+    },
+  );
+  if (loading && !fetchingMore) return <LoadingIndicator />;
+
+  if (error) {
+    throw new Error(error);
+  }
+
+  const loadingMoreMembers = networkStatus === 3;
+
+  let { members, cursor } = data.members.members;
```

Comment:
I would use state for these as well. I also bet Apollo is doing come caching of the results within it's store. Did you by chance play with that at all? Or happen to look at the `@connection` directive at all (https://www.apollographql.com/docs/react/data/pagination/#the-connection-directive)?

Reply:
nope, looked at how to know when loading on fetch more and this is what the people said to do.